### PR TITLE
 chore: refactor l10n

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: test/unit_tests/localization/LocalizerSpec.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ describe('l10n', () => {
           // no safe replacement
           expected: '<user>Felix</user>',
           params: {
-            substitutions: {replaceDangerously: {'\\/user': '</user>', user: '<user>'}},
+            substitutions: {replaceDangerously: {'/user': '</user>', user: '<user>'}},
```
Comment (atomrc) [inline test/unit_tests/localization/LocalizerSpec.js:138]: Could you just add the `number` test case for the `text` method, please. Then I think we are covered :+1:

File: app/script/localization/Localizer.js [hunk 2]
```diff
@@ -48,51 +48,42 @@ class Localizer {
...
-    });
-
-    return string;
+  const replaceSubstitute = (string, regex, substitute) => {
+    if (isStringOrNumber(substitute)) {
+      return string.replace(regex, substitute);
+    }
+    return string.replace(
+      regex,
+      (found, content) => (substitute.hasOwnProperty(content) ? substitute[content] : found)
```
Comment (gregor) [inline app/script/localization/Localizer.js]: Can we move this to a const `replaceFunction` so that the actual replace call turns out to be a oneliner?

Comment (AndyLnd) [reply app/script/localization/Localizer.js]: prettier strikes again ...

Comment (gregor) [reply app/script/localization/Localizer.js]: Or maybe even:
````
const replacement = isStringOrNumber(substitute)
  ? substitute
  : (found, content) => (substitute.hasOwnProperty(content) ? substitute[content] : found);+      
return string.replace(regex, replacement);
````

**Pull request merged**